tests/test_route_master.py

```diff
@@ -1,9 +1,51 @@
-from tests.util import TestCase
-
+from subway_structure import RouteMaster
 from tests.sample_data_for_twin_routes import metro_samples
+from tests.util import TestCase
 
 
 class TestRouteMaster(TestCase):
+    def test__find_common_circular_subsequence(self) -> None:
+        cases = [
+            {
+                "sequence1": [1, 2, 3, 4],
+                "sequence2": [1, 2, 3, 4],
+                "answer": [1, 2, 3, 4],
+            },
+            {
+                "sequence1": [1, 2, 3, 4],
+                "sequence2": [4, 1, 2, 3],
+                "answer": [1, 2, 3, 4],
+            },
+            {
+                "sequence1": [1, 2, 3, 4],
+                "sequence2": [2, 4, 1],
+                "answer": [1, 2, 4],
+            },
+            {
+                "sequence1": [1, 2, 4],
+                "sequence2": [2, 3, 4],
+                "answer": [2, 4],
+            },
+            {
+                "sequence1": [1, 2, 3, 4],
+                "sequence2": [5, 6, 7],
+                "answer": [],
+            },
+            {
+                "sequence1": [1, 2, 3, 4],
+                "sequence2": [4, 3, 2, 1],
+                "answer": [1, 2],
+            },
```

Comment:
I am not exactly sure how the function works, but here's more edge cases that I can think about just by looking at these ones:

1. `sequence1` is fully in `sequence2`;
2. `sequence1` = [], `sequence2` = [some numbers];
3. `sequence1` = [1, 2, 3, 4], `sequence2` = [4, 5, 6] (sequence2 starts from where sequence1 ends);

Reply:
Tests added. Thanks.